src/Store/MongoDB/Store.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Store\MongoDB;
+
+use MongoDB\BSON\Binary;
+use MongoDB\Client;
+use MongoDB\Collection;
+use PhpLlm\LlmChain\Document\Document;
+use PhpLlm\LlmChain\Document\Metadata;
+use PhpLlm\LlmChain\Document\Vector;
+use PhpLlm\LlmChain\Store\VectorStoreInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Uid\Uuid;
+
+/**
+ * @see https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-overview/
+ *
+ * For this store you need to create a separate MongoDB Atlas Search index.
+ * The index needs to be created with the following settings:
+ * {
+ *     "fields": [
+ *         {
+ *             "numDimensions": 1536,
+ *             "path": "vector",
+ *             "similarity": "euclidean",
+ *             "type": "vector"
+ *         }
+ *     ]
+ * }
+ *
+ * Note, that the `path` key needs to match the $vectorFieldName.
+ *
+ * For the `similarity` key you can choose between `euclidean`, `cosine` and `dotProduct`.
+ * {@see https://www.mongodb.com/docs/atlas/atlas-search/field-types/knn-vector/#define-the-index-for-the-fts-field-type-type}
+ *
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ */
+final readonly class Store implements VectorStoreInterface
+{
+    private Client $client;
+
+    /**
+     * @param string $uri             The URI to connect to the MongoDB server
+     * @param string $databaseName    The name of the database
+     * @param string $collectionName  The name of the collection
+     * @param string $indexName       The name of the Atlas Search index
+     * @param string $vectorFieldName The name of the field int the index that contains the vector
+     * @param bool   $bulkWrite       Use bulk write operations
+     */
+    public function __construct(
+        #[\SensitiveParameter] string $uri,
+        private LoggerInterface $logger,
+        private string $databaseName,
+        private string $collectionName,
+        private string $indexName,
+        private string $vectorFieldName = 'vector',
+        private bool $bulkWrite = false,
+    ) {
+        $this->client = new Client(
+            uri: $uri,
+            driverOptions: [
+                'driver' => [
+                    'name' => 'php-llm/llm-chain',
+                ],
+            ],
+        );
```

Comment:
Had the same issue with the third party client at https://github.com/php-llm/llm-chain/blob/main/src/Store/ChromaDB/Store.php, but decided for injection instead of instatiation because the client is a service in my perspective. but i ended up wiring that client myself.

what do you think about a factory approach instead?

Reply:
Sure I can go back to client injection.

Can you elaborate on the factory approach? How can that help here?

Reply:
the factory is basically an opinionated, but optional way to construct that store, which could be used by an implementation like our bundle, but this lib could be used in a different context where the client lifecycle is already handled differently and therefore injection would be more flexible.

Reply:
or maybe even the Client? :thinking:

Reply:
Let's go with the client for know, because when you install the Symfony mongodb bundle you are good to go ✌🏻

Reply:
isn't the mongodb bundle exactly an argument for injecting the Client?

so i would configure my client using that bundle, e.g.
```
mongodb:
  clients:
    default:
      uri: '%env(MONGODB_URI)%'
      default_database: #...
      uri_options: #...
      driver_options: #...
```

and than i would expect that this is sufficient to use that client as a vector store or do i get something wrong?

Reply:
> isn't the mongodb bundle exactly an argument for injecting the Client?

Yes